## Hypothesis 2: There are many experiments that will make the graph too dense and complex

We have stablished that there are experiments that repeat themselves. While I deleted all of the contradictory experiments, there are still repetitions. Let's see how much.


In [ ]:
# Count the number of unique experiments per dc_id and cell_line_id
groups = (
    final_experiments.groupby(["dc_id", "cell_line_id"])
    .filter(lambda x: len(x) >= 2)
    .groupby(["dc_id", "cell_line_id"])
    .size()
    .value_counts()
    .sort_index()
    .reset_index(name="count")
)
groups["total_size"] = groups["count"] * groups["index"]
groups["total_size"].sum()

np.int64(4903)

In [ ]:
groups["count"].sum()

np.int64(1547)

Seen this, we can see that 4903 experiments could be reduced to 1547 experiments. This will greatly reduce the complexity of the input graph for the GNN.

The question now is, how do I reduce this? I have though of three ways of doing it:

1. Average all the scores into one
2. Getting the geometric median of each score
3. Getting the max of each score (with the magnitude of the score vector)

However, as we have seen before, most of the repeated experiments are only repeated once, so using the IQR is impossible to determine outliers. So I'll focuse first on 2-repeated experiments and then the rest.

To go over this data, I will plot the difference between the data 
